src/encoder.rs

```diff
@@ -144,6 +137,11 @@ impl<W: Write> Encoder<W> {
         }.write_global_palette(global_palette)
     }
 
+    /// Write an extension block that signals a repeat behaviour.
+    pub fn set_repeat(&mut self, repeat: Repeat) -> Result<(), io::Error> {
+        self.write_extension(ExtensionData::Repetitions(repeat))
+    }
```

Comment:
I've gone with the pattern of using `set_` for all replacements but obviously there are other possibilities. For example,   `repeat` and `write_repeat`.